File: generated-code/python/example/TcpReadWrite/structure.py
from stream_wrapper import StreamWrapper

class Structure:
    """Example structure"""

    __slots__ = ("text","float_number","double_number",)

    text: str
    float_number: float
    double_number: float

    def __init__(self, text: str, float_number: float, double_number: float):
        self.text = text
        """Text"""
        self.float_number = float_number
        """32-bit float"""
        self.double_number = double_number
        """64-bit float"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "Structure":
        """Read Structure from input stream
        """
        text = stream.read_string()
        float_number = stream.read_float()
        double_number = stream.read_double()
        return Structure(text, float_number, double_number)
    
    def write_to(self, stream: StreamWrapper):
        """Write Structure to output stream
        """
        stream.write_string(self.text)
        stream.write_float(self.float_number)
        stream.write_double(self.double_number)
    
    def __repr__(self):
        return "Structure(" + \
            repr(self.text) + \
            ", " + \
            repr(self.float_number) + \
            ", " + \
            repr(self.double_number) + \
            ")"

File: generated-code/pascal/example/TcpReadWrite/Structure.pas
unit Structure;

interface

uses
    Stream,
    SysUtils;

type
    // Example structure
    TStructure = class
        // Text
        text: String;
        // 32-bit float
        floatNumber: Single;
        // 64-bit float
        doubleNumber: Double;
        constructor Create(text: String; floatNumber: Single; doubleNumber: Double);
        // Read Structure from input stream
        class function ReadFrom(stream: TStream): TStructure; static;
        // Write Structure to output stream
        procedure WriteTo(stream: TStream);
        function ToString: ansistring; override;
    end;

implementation

constructor TStructure.Create(text: String; floatNumber: Single; doubleNumber: Double);
begin
    self.text := text;
    self.floatNumber := floatNumber;
    self.doubleNumber := doubleNumber;
end;

class function TStructure.ReadFrom(stream: TStream): TStructure;
var doubleNumber: Double;
var floatNumber: Single;
var text: String;
begin
    text := stream.ReadString;
    floatNumber := stream.ReadSingle;
    doubleNumber := stream.ReadDouble;
    result := TStructure.Create(text, floatNumber, doubleNumber);
end;

procedure TStructure.WriteTo(stream: TStream);
begin
    stream.WriteString(text);
    stream.WriteSingle(floatNumber);
    stream.WriteDouble(doubleNumber);
end;

function TStructure.ToString: ansistring;
begin
    result := 'Structure {';
    result += 'text=';
    result += '''';
    result += text;
    result += '''';
    result += ', ';  
    result += 'floatNumber=';
    result += FloatToStr(floatNumber);
    result += ', ';  
    result += 'doubleNumber=';
    result += FloatToStr(doubleNumber);
    result += '}';
end;

end.